packages/manager/apps/dedicated/client/app/components/user/user.service.js

```diff
@@ -162,28 +146,10 @@ angular.module('services').service(
       });
     };
 
-    this.getValidPaymentMeansIds = function getValidPaymentMeansIds() {
-      const means = [
-        'bankAccount',
-        'paypal',
-        'creditCard',
-        'deferredPaymentAccount',
-      ];
-      const baseUrl = `${constants.swsProxyRootPath}me/paymentMean`;
-      const meanRequests = [];
-      means.forEach((paymentMethod) => {
-        let paramStruct = null;
-        if (paymentMethod === 'bankAccount') {
-          paramStruct = {
-            state: 'valid',
-          };
-        }
-        const promise = $http
-          .get([baseUrl, paymentMethod].join('/'), { params: paramStruct })
-          .then((response) => response.data);
-        meanRequests.push(promise);
-      });
-      return $q.all(meanRequests).then((response) => flatten(response));
+    this.getValidPaymentMethodIds = function getValidPaymentMethodIds() {
+      return $http
+        .get(`${constants.swsProxyRootPath}me/payment/method`)
+        .then((response) => response.data);
```

Comment:
You can directly do:
```suggestion
        .then(({ data }) => data);
```